config/routes.rb

```diff
@@ -142,6 +142,7 @@
     end
     namespace :users do
       resources :settings, only: %i[update]
+      resources :notification_settings, only: %i[update]
```

Comment:
I was thinking about this route, given that a user can only update their own setting and each user only has one row of notification settings, we don't really need to care about the ID of the row. So what if we switch it to a Rails [singular resource](https://guides.rubyonrails.org/routing.html#singular-resources)? 

This way we could have:

- `/users/notification_settings` as the path
- `users_notification_settings_path` as the helper

Reply:
I think I need some time to understand exactly how this works, before implementing

Reply:
@msarit The idea is basically that this implementation lets you make this request:

```http
PUT /users/notification_settings/1234
```

This makes sense if your notification setting id is `1234` or if you're an admin managing that notification setting for the user. But if neither of those are true, that URL path may not make sense. Best-case scenario is that the id is ignored. Worst-case is that it allows people to make requests regarding other people's settings.

@rhymes's suggestion removes the id requirement from the path (rather than requiring that it be provided only to throw it away). That way the user is derived from the session and the notification settings are derived from the user. No ids need to be provided:

```http
PUT /users/notification_settings
```

On the bright side, given the way you've built this, this looks like it'd be a one-character change — just gotta change `resources` to `resource` here. I don't see anything else that needs to change in this PR to accommodate it.

Reply:
Thank you for this excellent explanation @jgaskins! I appreciate you.